presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java

```diff
@@ -93,6 +98,29 @@ private boolean shouldUseTemporaryDirectory(Path path)
         }
     }
 
+    private Path createTemporaryPath(HdfsEnvironment hdfsEnvironment, Path targetPath)
+    {
+        // use a per-user temporary directory to avoid permission problems
+
+        String temporaryPrefix;
+        try {
+            String currentUser = UserGroupInformation.getCurrentUser().getUserName();
+            temporaryPrefix = hiveClientConfig.getTemporaryDirectory().replace("%USER%", currentUser);
```

Comment:
maybe getter should accept `userName` parameters, so semantic of property could be enclosed in configuration class